### ISSUE TITLE
Crash ``Building error when trying to create ast representation of module``

### ISSUE THREAD

**Issue by nachomaiz** [pylint-dev/pylint#9116]
### Bug description

```python
The crash happens both when using pylint from the command line and the extension for VSCode (including pre-release) **with Python 3.12**.

Code might not necessarily be relevant since a similar error happens with every file in the [repository](https://github.com/wppbav/bavapi-sdk-python/)

I think I've narrowed it down to any file that imports `typing` at any point in the import chain.

So, for example this doesn't trigger the error:


import argparse

def main(argv: list[str]) -> None:
    parser = argparse.ArgumentParser(argv)


But this does:
```py
import argparse
from typing import List  # used for py3.8 support

def main(argv: List[str]) -> None:
    parser = argparse.ArgumentParser(argv)
```

A file that imports that main function would also trigger the error.

I do use `TypeAlias` (not the PEP 695 version for backwards compatibility) in various modules, but even those with no `TypeAlias` have the same issue. The (I believe relevant) exception that I get from all files is the same:

```
'TreeRebuilder' object has no attribute 'visit_typealias'
```

So, since I use the typing module across all files, all files are triggering the error.

Here's the stack trace (user path obscured with `~`):

pylint crashed with a ``AstroidBuildingError`` and with the following stacktrace:
```
Traceback (most recent call last):
  File "~\.vscode\extensions\ms-python.pylint-2023.7.12711014\bundled\libs\pylint\lint\pylinter.py", line 1045, in get_ast
    return astroid.builder.AstroidBuilder(MANAGER).string_build(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "~\.vscode\extensions\ms-python.pylint-2023.7.12711014\bundled\libs\astroid\builder.py", line 153, in string_build
    return self._post_build(module, builder, "utf-8")
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "~\.vscode\extensions\ms-python.pylint-2023.7.12711014\bundled\libs\astroid\builder.py", line 173, in _post_build
    module = self._manager.visit_transforms(module)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "~\.vscode\extensions\ms-python.pylint-2023.7.12711014\bundled\libs\astroid\manager.py", line 109, in visit_transforms
    return self._transform.visit(node)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "~\.vscode\extensions\ms-python.pylint-2023.7.12711014\bundled\libs\astroid\transforms.py", line 89, in visit
    return self._visit(module)
           ^^^^^^^^^^^^^^^^^^^
  File "~\.vscode\extensions\ms-python.pylint-2023.7.12711014\bundled\libs\astroid\transforms.py", line 54, in _visit
    visited = self._visit_generic(value)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "~\.vscode\extensions\ms-python.pylint-2023.7.12711014\bundled\libs\astroid\transforms.py", line 61, in _visit_generic
    return [self._visit_generic(child) for child in node]
            ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "~\.vscode\extensions\ms-python.pylint-2023.7.12711014\bundled\libs\astroid\transforms.py", line 67, in _visit_generic
    return self._visit(node)
           ^^^^^^^^^^^^^^^^^
  File "~\.vscode\extensions\ms-python.pylint-2023.7.12711014\bundled\libs\astroid\transforms.py", line 57, in _visit
    return self._transform(node)
           ^^^^^^^^^^^^^^^^^^^^^
  File "~\.vscode\extensions\ms-python.pylint-2023.7.12711014\bundled\libs\astroid\transforms.py", line 38, in _transform
    if predicate is None or predicate(node):
                            ^^^^^^^^^^^^^^^
  File "~\.vscode\extensions\ms-python.pylint-2023.7.12711014\bundled\libs\astroid\brain\brain_namedtuple_enum.py", line 613, in _is_enum_subclass
    for klass in cls.mro()
                 ^^^^^^^^^
  File "~\.vscode\extensions\ms-python.pylint-2023.7.12711014\bundled\libs\astroid\nodes\scoped_nodes\scoped_nodes.py", line 3066, in mro
    return self._compute_mro(context=context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "~\.vscode\extensions\ms-python.pylint-2023.7.12711014\bundled\libs\astroid\nodes\scoped_nodes\scoped_nodes.py", line 3035, in _compute_mro
    inferred_bases = list(self._inferred_bases(context=context))
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "~\.vscode\extensions\ms-python.pylint-2023.7.12711014\bundled\libs\astroid\nodes\scoped_nodes\scoped_nodes.py", line 3018, in _inferred_bases
    baseobj = next(
              ^^^^^
  File "~\.vscode\extensions\ms-python.pylint-2023.7.12711014\bundled\libs\astroid\nodes\scoped_nodes\scoped_nodes.py", line 3018, in <genexpr>
    baseobj = next(
                  ^
  File "~\.vscode\extensions\ms-python.pylint-2023.7.12711014\bundled\libs\astroid\nodes\node_ng.py", line 184, in infer
    for i, result in enumerate(self._infer(context=context, **kwargs)):
  File "~\.vscode\extensions\ms-python.pylint-2023.7.12711014\bundled\libs\astroid\decorators.py", line 143, in raise_if_nothing_inferred
    yield next(generator)
          ^^^^^^^^^^^^^^^
  File "~\.vscode\extensions\ms-python.pylint-2023.7.12711014\bundled\libs\astroid\decorators.py", line 112, in wrapped
    for res in _func(node, context, **kwargs):
  File "~\.vscode\extensions\ms-python.pylint-2023.7.12711014\bundled\libs\astroid\bases.py", line 177, in _infer_stmts
    for inf in stmt.infer(context=context):
  File "~\.vscode\extensions\ms-python.pylint-2023.7.12711014\bundled\libs\astroid\nodes\node_ng.py", line 184, in infer
    for i, result in enumerate(self._infer(context=context, **kwargs)):
  File "~\.vscode\extensions\ms-python.pylint-2023.7.12711014\bundled\libs\astroid\decorators.py", line 143, in raise_if_nothing_inferred
    yield next(generator)
          ^^^^^^^^^^^^^^^
  File "~\.vscode\extensions\ms-python.pylint-2023.7.12711014\bundled\libs\astroid\decorators.py", line 112, in wrapped
    for res in _func(node, context, **kwargs):
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "~\.vscode\extensions\ms-python.pylint-2023.7.12711014\bundled\libs\astroid\inference.py", line 334, in infer_import_from
    module = self.do_import_module()
             ^^^^^^^^^^^^^^^^^^^^^^^
  File "~\.vscode\extensions\ms-python.pylint-2023.7.12711014\bundled\libs\astroid\nodes\_base_nodes.py", line 146, in do_import_module
    return mymodule.import_module(
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "~\.vscode\extensions\ms-python.pylint-2023.7.12711014\bundled\libs\astroid\nodes\scoped_nodes\scoped_nodes.py", line 530, in import_module
    return AstroidManager().ast_from_module_name(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "~\.vscode\extensions\ms-python.pylint-2023.7.12711014\bundled\libs\astroid\manager.py", line 246, in ast_from_module_name
    return self.ast_from_file(found_spec.location, modname, fallback=False)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "~\.vscode\extensions\ms-python.pylint-2023.7.12711014\bundled\libs\astroid\manager.py", line 138, in ast_from_file
    return AstroidBuilder(self).file_build(filepath, modname)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "~\.vscode\extensions\ms-python.pylint-2023.7.12711014\bundled\libs\astroid\builder.py", line 145, in file_build
    return self._post_build(module, builder, encoding)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "~\.vscode\extensions\ms-python.pylint-2023.7.12711014\bundled\libs\astroid\builder.py", line 166, in _post_build
    self.add_from_names_to_locals(from_node)
  File "~\.vscode\extensions\ms-python.pylint-2023.7.12711014\bundled\libs\astroid\builder.py", line 223, in add_from_names_to_locals
    imported = node.do_import_module()
               ^^^^^^^^^^^^^^^^^^^^^^^
  File "~\.vscode\extensions\ms-python.pylint-2023.7.12711014\bundled\libs\astroid\nodes\_base_nodes.py", line 146, in do_import_module
    return mymodule.import_module(
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "~\.vscode\extensions\ms-python.pylint-2023.7.12711014\bundled\libs\astroid\nodes\scoped_nodes\scoped_nodes.py", line 530, in import_module
    return AstroidManager().ast_from_module_name(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "~\.vscode\extensions\ms-python.pylint-2023.7.12711014\bundled\libs\astroid\manager.py", line 246, in ast_from_module_name
    return self.ast_from_file(found_spec.location, modname, fallback=False)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "~\.vscode\extensions\ms-python.pylint-2023.7.12711014\bundled\libs\astroid\manager.py", line 138, in ast_from_file
    return AstroidBuilder(self).file_build(filepath, modname)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "~\.vscode\extensions\ms-python.pylint-2023.7.12711014\bundled\libs\astroid\builder.py", line 145, in file_build
    return self._post_build(module, builder, encoding)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "~\.vscode\extensions\ms-python.pylint-2023.7.12711014\bundled\libs\astroid\builder.py", line 173, in _post_build
    module = self._manager.visit_transforms(module)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "~\.vscode\extensions\ms-python.pylint-2023.7.12711014\bundled\libs\astroid\manager.py", line 109, in visit_transforms
    return self._transform.visit(node)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "~\.vscode\extensions\ms-python.pylint-2023.7.12711014\bundled\libs\astroid\transforms.py", line 89, in visit
    return self._visit(module)
           ^^^^^^^^^^^^^^^^^^^
  File "~\.vscode\extensions\ms-python.pylint-2023.7.12711014\bundled\libs\astroid\transforms.py", line 54, in _visit
    visited = self._visit_generic(value)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "~\.vscode\extensions\ms-python.pylint-2023.7.12711014\bundled\libs\astroid\transforms.py", line 61, in _visit_generic
    return [self._visit_generic(child) for child in node]
            ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "~\.vscode\extensions\ms-python.pylint-2023.7.12711014\bundled\libs\astroid\transforms.py", line 67, in _visit_generic
    return self._visit(node)
           ^^^^^^^^^^^^^^^^^
  File "~\.vscode\extensions\ms-python.pylint-2023.7.12711014\bundled\libs\astroid\transforms.py", line 57, in _visit
    return self._transform(node)
           ^^^^^^^^^^^^^^^^^^^^^
  File "~\.vscode\extensions\ms-python.pylint-2023.7.12711014\bundled\libs\astroid\transforms.py", line 39, in _transform
    ret = transform_func(node)
          ^^^^^^^^^^^^^^^^^^^^
  File "~\.vscode\extensions\ms-python.pylint-2023.7.12711014\bundled\libs\astroid\brain\brain_dataclasses.py", line 90, in dataclass_transform
    init_str = _generate_dataclass_init(
               ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "~\.vscode\extensions\ms-python.pylint-2023.7.12711014\bundled\libs\astroid\brain\brain_dataclasses.py", line 255, in _generate_dataclass_init
    prev_pos_only_store, prev_kw_only_store = _find_arguments_from_base_classes(node)
                                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "~\.vscode\extensions\ms-python.pylint-2023.7.12711014\bundled\libs\astroid\brain\brain_dataclasses.py", line 180, in _find_arguments_from_base_classes
    for base in reversed(node.mro()):
                         ^^^^^^^^^^
  File "~\.vscode\extensions\ms-python.pylint-2023.7.12711014\bundled\libs\astroid\nodes\scoped_nodes\scoped_nodes.py", line 3066, in mro
    return self._compute_mro(context=context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "~\.vscode\extensions\ms-python.pylint-2023.7.12711014\bundled\libs\astroid\nodes\scoped_nodes\scoped_nodes.py", line 3035, in _compute_mro
    inferred_bases = list(self._inferred_bases(context=context))
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "~\.vscode\extensions\ms-python.pylint-2023.7.12711014\bundled\libs\astroid\nodes\scoped_nodes\scoped_nodes.py", line 3018, in _inferred_bases
    baseobj = next(
              ^^^^^
  File "~\.vscode\extensions\ms-python.pylint-2023.7.12711014\bundled\libs\astroid\nodes\scoped_nodes\scoped_nodes.py", line 3018, in <genexpr>
    baseobj = next(
                  ^
  File "~\.vscode\extensions\ms-python.pylint-2023.7.12711014\bundled\libs\astroid\nodes\node_ng.py", line 184, in infer
    for i, result in enumerate(self._infer(context=context, **kwargs)):
  File "~\.vscode\extensions\ms-python.pylint-2023.7.12711014\bundled\libs\astroid\decorators.py", line 143, in raise_if_nothing_inferred
    yield next(generator)
          ^^^^^^^^^^^^^^^
  File "~\.vscode\extensions\ms-python.pylint-2023.7.12711014\bundled\libs\astroid\decorators.py", line 112, in wrapped
    for res in _func(node, context, **kwargs):
  File "~\.vscode\extensions\ms-python.pylint-2023.7.12711014\bundled\libs\astroid\inference.py", line 358, in infer_attribute
    for owner in self.expr.infer(context):
  File "~\.vscode\extensions\ms-python.pylint-2023.7.12711014\bundled\libs\astroid\nodes\node_ng.py", line 184, in infer
    for i, result in enumerate(self._infer(context=context, **kwargs)):
  File "~\.vscode\extensions\ms-python.pylint-2023.7.12711014\bundled\libs\astroid\decorators.py", line 143, in raise_if_nothing_inferred
    yield next(generator)
          ^^^^^^^^^^^^^^^
  File "~\.vscode\extensions\ms-python.pylint-2023.7.12711014\bundled\libs\astroid\decorators.py", line 112, in wrapped
    for res in _func(node, context, **kwargs):
  File "~\.vscode\extensions\ms-python.pylint-2023.7.12711014\bundled\libs\astroid\bases.py", line 177, in _infer_stmts
    for inf in stmt.infer(context=context):
  File "~\.vscode\extensions\ms-python.pylint-2023.7.12711014\bundled\libs\astroid\nodes\node_ng.py", line 184, in infer
    for i, result in enumerate(self._infer(context=context, **kwargs)):
  File "~\.vscode\extensions\ms-python.pylint-2023.7.12711014\bundled\libs\astroid\decorators.py", line 143, in raise_if_nothing_inferred
    yield next(generator)
          ^^^^^^^^^^^^^^^
  File "~\.vscode\extensions\ms-python.pylint-2023.7.12711014\bundled\libs\astroid\decorators.py", line 112, in wrapped
    for res in _func(node, context, **kwargs):
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "~\.vscode\extensions\ms-python.pylint-2023.7.12711014\bundled\libs\astroid\inference.py", line 341, in infer_import_from
    stmts = module.getattr(name, ignore_locals=module is self.root())
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "~\.vscode\extensions\ms-python.pylint-2023.7.12711014\bundled\libs\astroid\nodes\scoped_nodes\scoped_nodes.py", line 414, in getattr
    result = [self.import_module(name, relative_only=True)]
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "~\.vscode\extensions\ms-python.pylint-2023.7.12711014\bundled\libs\astroid\nodes\scoped_nodes\scoped_nodes.py", line 530, in import_module
    return AstroidManager().ast_from_module_name(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "~\.vscode\extensions\ms-python.pylint-2023.7.12711014\bundled\libs\astroid\manager.py", line 246, in ast_from_module_name
    return self.ast_from_file(found_spec.location, modname, fallback=False)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "~\.vscode\extensions\ms-python.pylint-2023.7.12711014\bundled\libs\astroid\manager.py", line 138, in ast_from_file
    return AstroidBuilder(self).file_build(filepath, modname)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "~\.vscode\extensions\ms-python.pylint-2023.7.12711014\bundled\libs\astroid\builder.py", line 145, in file_build
    return self._post_build(module, builder, encoding)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "~\.vscode\extensions\ms-python.pylint-2023.7.12711014\bundled\libs\astroid\builder.py", line 169, in _post_build
    self.delayed_assattr(delayed)
  File "~\.vscode\extensions\ms-python.pylint-2023.7.12711014\bundled\libs\astroid\builder.py", line 240, in delayed_assattr
    for inferred in node.expr.infer():
  File "~\.vscode\extensions\ms-python.pylint-2023.7.12711014\bundled\libs\astroid\nodes\node_ng.py", line 171, in infer
    yield from self._infer(context=context, **kwargs)
  File "~\.vscode\extensions\ms-python.pylint-2023.7.12711014\bundled\libs\astroid\decorators.py", line 143, in raise_if_nothing_inferred
    yield next(generator)
          ^^^^^^^^^^^^^^^
  File "~\.vscode\extensions\ms-python.pylint-2023.7.12711014\bundled\libs\astroid\decorators.py", line 112, in wrapped
    for res in _func(node, context, **kwargs):
  File "~\.vscode\extensions\ms-python.pylint-2023.7.12711014\bundled\libs\astroid\bases.py", line 177, in _infer_stmts
    for inf in stmt.infer(context=context):
  File "~\.vscode\extensions\ms-python.pylint-2023.7.12711014\bundled\libs\astroid\nodes\node_ng.py", line 184, in infer
    for i, result in enumerate(self._infer(context=context, **kwargs)):
  File "~\.vscode\extensions\ms-python.pylint-2023.7.12711014\bundled\libs\astroid\decorators.py", line 143, in raise_if_nothing_inferred
    yield next(generator)
          ^^^^^^^^^^^^^^^
  File "~\.vscode\extensions\ms-python.pylint-2023.7.12711014\bundled\libs\astroid\decorators.py", line 112, in wrapped
    for res in _func(node, context, **kwargs):
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "~\.vscode\extensions\ms-python.pylint-2023.7.12711014\bundled\libs\astroid\inference.py", line 1157, in infer_assign
    stmts = list(self.assigned_stmts(context=context))
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "~\.vscode\extensions\ms-python.pylint-2023.7.12711014\bundled\libs\astroid\protocols.py", line 412, in _arguments_infer_argname
    is_metaclass = isinstance(cls, nodes.ClassDef) and cls.type == "metaclass"
                                                       ^^^^^^^^
  File "~\.vscode\extensions\ms-python.pylint-2023.7.12711014\bundled\libs\astroid\nodes\scoped_nodes\scoped_nodes.py", line 1874, in _class_type
    if _is_metaclass(klass):
       ^^^^^^^^^^^^^^^^^^^^
  File "~\.vscode\extensions\ms-python.pylint-2023.7.12711014\bundled\libs\astroid\nodes\scoped_nodes\scoped_nodes.py", line 1845, in _is_metaclass
    for baseobj in base.infer():
  File "~\.vscode\extensions\ms-python.pylint-2023.7.12711014\bundled\libs\astroid\nodes\node_ng.py", line 171, in infer
    yield from self._infer(context=context, **kwargs)
  File "~\.vscode\extensions\ms-python.pylint-2023.7.12711014\bundled\libs\astroid\decorators.py", line 143, in raise_if_nothing_inferred
    yield next(generator)
          ^^^^^^^^^^^^^^^
  File "~\.vscode\extensions\ms-python.pylint-2023.7.12711014\bundled\libs\astroid\decorators.py", line 112, in wrapped
    for res in _func(node, context, **kwargs):
  File "~\.vscode\extensions\ms-python.pylint-2023.7.12711014\bundled\libs\astroid\bases.py", line 177, in _infer_stmts
    for inf in stmt.infer(context=context):
  File "~\.vscode\extensions\ms-python.pylint-2023.7.12711014\bundled\libs\astroid\nodes\node_ng.py", line 184, in infer
    for i, result in enumerate(self._infer(context=context, **kwargs)):
  File "~\.vscode\extensions\ms-python.pylint-2023.7.12711014\bundled\libs\astroid\decorators.py", line 143, in raise_if_nothing_inferred
    yield next(generator)
          ^^^^^^^^^^^^^^^
  File "~\.vscode\extensions\ms-python.pylint-2023.7.12711014\bundled\libs\astroid\decorators.py", line 112, in wrapped
    for res in _func(node, context, **kwargs):
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "~\.vscode\extensions\ms-python.pylint-2023.7.12711014\bundled\libs\astroid\inference.py", line 334, in infer_import_from
    module = self.do_import_module()
             ^^^^^^^^^^^^^^^^^^^^^^^
  File "~\.vscode\extensions\ms-python.pylint-2023.7.12711014\bundled\libs\astroid\nodes\_base_nodes.py", line 146, in do_import_module
    return mymodule.import_module(
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "~\.vscode\extensions\ms-python.pylint-2023.7.12711014\bundled\libs\astroid\nodes\scoped_nodes\scoped_nodes.py", line 530, in import_module
    return AstroidManager().ast_from_module_name(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "~\.vscode\extensions\ms-python.pylint-2023.7.12711014\bundled\libs\astroid\manager.py", line 246, in ast_from_module_name
    return self.ast_from_file(found_spec.location, modname, fallback=False)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "~\.vscode\extensions\ms-python.pylint-2023.7.12711014\bundled\libs\astroid\manager.py", line 138, in ast_from_file
    return AstroidBuilder(self).file_build(filepath, modname)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "~\.vscode\extensions\ms-python.pylint-2023.7.12711014\bundled\libs\astroid\builder.py", line 145, in file_build
    return self._post_build(module, builder, encoding)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "~\.vscode\extensions\ms-python.pylint-2023.7.12711014\bundled\libs\astroid\builder.py", line 173, in _post_build
    module = self._manager.visit_transforms(module)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "~\.vscode\extensions\ms-python.pylint-2023.7.12711014\bundled\libs\astroid\manager.py", line 109, in visit_transforms
    return self._transform.visit(node)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "~\.vscode\extensions\ms-python.pylint-2023.7.12711014\bundled\libs\astroid\transforms.py", line 89, in visit
    return self._visit(module)
           ^^^^^^^^^^^^^^^^^^^
  File "~\.vscode\extensions\ms-python.pylint-2023.7.12711014\bundled\libs\astroid\transforms.py", line 54, in _visit
    visited = self._visit_generic(value)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "~\.vscode\extensions\ms-python.pylint-2023.7.12711014\bundled\libs\astroid\transforms.py", line 61, in _visit_generic
    return [self._visit_generic(child) for child in node]
            ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "~\.vscode\extensions\ms-python.pylint-2023.7.12711014\bundled\libs\astroid\transforms.py", line 67, in _visit_generic
    return self._visit(node)
           ^^^^^^^^^^^^^^^^^
  File "~\.vscode\extensions\ms-python.pylint-2023.7.12711014\bundled\libs\astroid\transforms.py", line 57, in _visit
    return self._transform(node)
           ^^^^^^^^^^^^^^^^^^^^^
  File "~\.vscode\extensions\ms-python.pylint-2023.7.12711014\bundled\libs\astroid\transforms.py", line 38, in _transform
    if predicate is None or predicate(node):
                            ^^^^^^^^^^^^^^^
  File "~\.vscode\extensions\ms-python.pylint-2023.7.12711014\bundled\libs\astroid\brain\brain_namedtuple_enum.py", line 613, in _is_enum_subclass
    for klass in cls.mro()
                 ^^^^^^^^^
  File "~\.vscode\extensions\ms-python.pylint-2023.7.12711014\bundled\libs\astroid\nodes\scoped_nodes\scoped_nodes.py", line 3066, in mro
    return self._compute_mro(context=context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "~\.vscode\extensions\ms-python.pylint-2023.7.12711014\bundled\libs\astroid\nodes\scoped_nodes\scoped_nodes.py", line 3035, in _compute_mro
    inferred_bases = list(self._inferred_bases(context=context))
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "~\.vscode\extensions\ms-python.pylint-2023.7.12711014\bundled\libs\astroid\nodes\scoped_nodes\scoped_nodes.py", line 3018, in _inferred_bases
    baseobj = next(
              ^^^^^
  File "~\.vscode\extensions\ms-python.pylint-2023.7.12711014\bundled\libs\astroid\nodes\scoped_nodes\scoped_nodes.py", line 3018, in <genexpr>
    baseobj = next(
                  ^
  File "~\.vscode\extensions\ms-python.pylint-2023.7.12711014\bundled\libs\astroid\nodes\node_ng.py", line 184, in infer
    for i, result in enumerate(self._infer(context=context, **kwargs)):
  File "~\.vscode\extensions\ms-python.pylint-2023.7.12711014\bundled\libs\astroid\decorators.py", line 143, in raise_if_nothing_inferred
    yield next(generator)
          ^^^^^^^^^^^^^^^
  File "~\.vscode\extensions\ms-python.pylint-2023.7.12711014\bundled\libs\astroid\decorators.py", line 112, in wrapped
    for res in _func(node, context, **kwargs):
  File "~\.vscode\extensions\ms-python.pylint-2023.7.12711014\bundled\libs\astroid\bases.py", line 177, in _infer_stmts
    for inf in stmt.infer(context=context):
  File "~\.vscode\extensions\ms-python.pylint-2023.7.12711014\bundled\libs\astroid\nodes\node_ng.py", line 184, in infer
    for i, result in enumerate(self._infer(context=context, **kwargs)):
  File "~\.vscode\extensions\ms-python.pylint-2023.7.12711014\bundled\libs\astroid\decorators.py", line 143, in raise_if_nothing_inferred
    yield next(generator)
          ^^^^^^^^^^^^^^^
  File "~\.vscode\extensions\ms-python.pylint-2023.7.12711014\bundled\libs\astroid\decorators.py", line 112, in wrapped
    for res in _func(node, context, **kwargs):
  File "~\.vscode\extensions\ms-python.pylint-2023.7.12711014\bundled\libs\astroid\bases.py", line 177, in _infer_stmts
    for inf in stmt.infer(context=context):
  File "~\.vscode\extensions\ms-python.pylint-2023.7.12711014\bundled\libs\astroid\nodes\node_ng.py", line 184, in infer
    for i, result in enumerate(self._infer(context=context, **kwargs)):
  File "~\.vscode\extensions\ms-python.pylint-2023.7.12711014\bundled\libs\astroid\decorators.py", line 143, in raise_if_nothing_inferred
    yield next(generator)
          ^^^^^^^^^^^^^^^
  File "~\.vscode\extensions\ms-python.pylint-2023.7.12711014\bundled\libs\astroid\decorators.py", line 112, in wrapped
    for res in _func(node, context, **kwargs):
  File "~\.vscode\extensions\ms-python.pylint-2023.7.12711014\bundled\libs\astroid\bases.py", line 177, in _infer_stmts
    for inf in stmt.infer(context=context):
  File "~\.vscode\extensions\ms-python.pylint-2023.7.12711014\bundled\libs\astroid\nodes\node_ng.py", line 184, in infer
    for i, result in enumerate(self._infer(context=context, **kwargs)):
  File "~\.vscode\extensions\ms-python.pylint-2023.7.12711014\bundled\libs\astroid\decorators.py", line 143, in raise_if_nothing_inferred
    yield next(generator)
          ^^^^^^^^^^^^^^^
  File "~\.vscode\extensions\ms-python.pylint-2023.7.12711014\bundled\libs\astroid\decorators.py", line 112, in wrapped
    for res in _func(node, context, **kwargs):
  File "~\.vscode\extensions\ms-python.pylint-2023.7.12711014\bundled\libs\astroid\inference.py", line 358, in infer_attribute
    for owner in self.expr.infer(context):
  File "~\.vscode\extensions\ms-python.pylint-2023.7.12711014\bundled\libs\astroid\nodes\node_ng.py", line 184, in infer
    for i, result in enumerate(self._infer(context=context, **kwargs)):
  File "~\.vscode\extensions\ms-python.pylint-2023.7.12711014\bundled\libs\astroid\decorators.py", line 143, in raise_if_nothing_inferred
    yield next(generator)
          ^^^^^^^^^^^^^^^
  File "~\.vscode\extensions\ms-python.pylint-2023.7.12711014\bundled\libs\astroid\decorators.py", line 112, in wrapped
    for res in _func(node, context, **kwargs):
  File "~\.vscode\extensions\ms-python.pylint-2023.7.12711014\bundled\libs\astroid\bases.py", line 177, in _infer_stmts
    for inf in stmt.infer(context=context):
  File "~\.vscode\extensions\ms-python.pylint-2023.7.12711014\bundled\libs\astroid\nodes\node_ng.py", line 184, in infer
    for i, result in enumerate(self._infer(context=context, **kwargs)):
  File "~\.vscode\extensions\ms-python.pylint-2023.7.12711014\bundled\libs\astroid\decorators.py", line 143, in raise_if_nothing_inferred
    yield next(generator)
          ^^^^^^^^^^^^^^^
  File "~\.vscode\extensions\ms-python.pylint-2023.7.12711014\bundled\libs\astroid\decorators.py", line 112, in wrapped
    for res in _func(node, context, **kwargs):
  File "~\.vscode\extensions\ms-python.pylint-2023.7.12711014\bundled\libs\astroid\inference.py", line 304, in infer_import
    yield self.do_import_module(self.real_name(name))
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "~\.vscode\extensions\ms-python.pylint-2023.7.12711014\bundled\libs\astroid\nodes\_base_nodes.py", line 146, in do_import_module
    return mymodule.import_module(
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "~\.vscode\extensions\ms-python.pylint-2023.7.12711014\bundled\libs\astroid\nodes\scoped_nodes\scoped_nodes.py", line 530, in import_module
    return AstroidManager().ast_from_module_name(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "~\.vscode\extensions\ms-python.pylint-2023.7.12711014\bundled\libs\astroid\manager.py", line 246, in ast_from_module_name
    return self.ast_from_file(found_spec.location, modname, fallback=False)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "~\.vscode\extensions\ms-python.pylint-2023.7.12711014\bundled\libs\astroid\manager.py", line 138, in ast_from_file
    return AstroidBuilder(self).file_build(filepath, modname)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "~\.vscode\extensions\ms-python.pylint-2023.7.12711014\bundled\libs\astroid\builder.py", line 144, in file_build
    module, builder = self._data_build(data, modname, path)
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "~\.vscode\extensions\ms-python.pylint-2023.7.12711014\bundled\libs\astroid\builder.py", line 204, in _data_build
    module = builder.visit_module(node, modname, node_file, package)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "~\.vscode\extensions\ms-python.pylint-2023.7.12711014\bundled\libs\astroid\rebuilder.py", line 254, in visit_module
    [self.visit(child, newnode) for child in node.body],
     ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "~\.vscode\extensions\ms-python.pylint-2023.7.12711014\bundled\libs\astroid\rebuilder.py", line 609, in visit
    visit_method = getattr(self, visit_name)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'TreeRebuilder' object has no attribute 'visit_typealias'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "~\.vscode\extensions\ms-python.pylint-2023.7.12711014\bundled\libs\pylint\lint\pylinter.py", line 730, in _get_asts
    ast_per_fileitem[fileitem] = self.get_ast(
                                 ^^^^^^^^^^^^^
  File "~\.vscode\extensions\ms-python.pylint-2023.7.12711014\bundled\libs\pylint\lint\pylinter.py", line 1066, in get_ast
    raise astroid.AstroidBuildingError(
astroid.exceptions.AstroidBuildingError: Building error when trying to create ast representation of module 'bavapi.filters'
```
```


### Configuration

_No response_

### Command used

```shell
pylint .

Or from automatic linting by the VSCode extension
```

### Pylint output

```python
Example from one of the files in the repository

bavapi\parsing\responses.py:1:0: F0002: bavapi\parsing\responses.py: Fatal error while checking 'bavapi\parsing\responses.py'. Please open an issue in our bug tracker so we address this. There is a pre-filled template that you can use in '~\AppData\Local\pylint\pylint\Cache\pylint-crash-2023-10-04-14-28-37.txt'. (astroid-error)
```

### Expected behavior

Pylint should be able to parse these files, especially since I haven't introduced PEP 695 style type aliases to the codebase.

### Pylint version

```shell
From command line:
pylint 2.17.4
astroid 2.15.5
Python 3.12.0 | packaged by conda-forge | (main, Oct  3 2023, 08:26:13) [MSC v.1935 64 bit (AMD64)]

Extension:
Pylint: v2023.7.12711014
Pylance: v2023.9.31
```

### OS / Environment

Windows 10 x64

**Comment by jacobtylerwalls** [comment]
Hi 👋🏻. The issue here is that pylint 2.x is not compatible with Python 3.12. You can still use 3.11 and lower.

**Comment by jacobtylerwalls** [comment]
(Duplicate of #9091)
